unable to build lib on Mac M1

I tried to build library for iOS using tutorial from README file but I am getting following error in xCode:

running prepare-cyrus-sasl.sh
prepare sources
building tools
generated makemd5arm64 properly
building for iPhoneOS - armv7
log to /libetpan/build-mac/dependencies/build/libsasl/workdir/20230516192856/src/cyrus-sasl-2.1.28-iPhoneOS-armv7/build.log
Command ExternalBuildToolExecution failed with a nonzero exit code

and here is a error from log file mentioned above:

*** building for iPhoneOS - armv7 ***
configure: WARNING: you should use --build, --host, --target
configure: creating cache ./config.cache
checking build system type... Invalid configuration `16.4armv7': machine `16.4armv7' not recognized
configure: error: /bin/sh config/config.sub 16.4armv7 failed

--- a/build-mac/dependencies/prepare-cyrus-sasl.sh
+++ b/build-mac/dependencies/prepare-cyrus-sasl.sh
@@ -115,18 +115,22 @@ export LANG=en_US.US-ASCII
 LIB_NAME=$ARCHIVE
 TARGETS="iPhoneOS iPhoneSimulator"
 
-SDK_IOS_MIN_VERSION=7.0
+SDK_IOS_MIN_VERSION=15.0
 SDK_IOS_VERSION="`xcodebuild -showsdks 2>/dev/null | grep iphoneos | sed 's/.*iphoneos\(.*\)/\1/'`"
 BUILD_DIR="$tmpdir/build"
 INSTALL_PATH="${BUILD_DIR}/${LIB_NAME}/universal"
-BITCODE_FLAGS="-fembed-bitcode"
-if test "x$NOBITCODE" != x ; then
-   BITCODE_FLAGS=""
-fi
+BITCODE_FLAGS=""
 
 xcode_developer="$(xcode-select -p)"
 
 function build_target {
+  local TARGET="$1"
+  local MARCH="$2" 
+  local ARCH="$3"
+  local SYSROOT="$4"
+  local EXTRA_FLAGS="$5"
+  local TARGET_SUFFIX="$6"
+  
   local current_logfile="$srcdir/$ARCHIVE-$TARGET-$MARCH/build.log"
   echo "log to $current_logfile"
   cp -R "$srcdir/$ARCHIVE" "$srcdir/$ARCHIVE-$TARGET-$MARCH"
@@ -140,7 +144,7 @@ function build_target {
   
   local CURRENT_TARGET="$MARCH-apple-ios${SDK_IOS_MIN_VERSION}${TARGET_SUFFIX}"
   export CPPFLAGS="-isysroot ${SYSROOT} -target ${CURRENT_TARGET} -Os"
-  export CFLAGS="${CPPFLAGS} ${EXTRA_FLAGS}"
+  export CFLAGS="-isysroot ${SYSROOT} -target ${CURRENT_TARGET} -Os ${EXTRA_FLAGS}"
   export CC="$xcode_developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang"
   export CXX="$xcode_developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++"
   export LD="$xcode_developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld"
@@ -178,20 +182,24 @@ pids=""
 for TARGET in $TARGETS; do
 
     DEVELOPER="$(xcode-select --print-path)"
-    SDK_ID="$(echo "$TARGET$SDK_IOS_VERSION" | tr A-Z a-z)"
+    if [ "$TARGET" = "iPhoneOS" ]; then
+        SDK_ID="iphoneos$SDK_IOS_VERSION"
+    else
+        SDK_ID="iphonesimulator$SDK_IOS_VERSION"
+    fi
     SYSROOT="$(xcodebuild -version -sdk "$SDK_ID" 2>/dev/null | egrep '^Path: ' | cut -d ' ' -f 2)"
 
     case $TARGET in
         (iPhoneOS)
             ARCH=arm
-            MARCHS="armv7 armv7s arm64"
+            MARCHS="arm64"
             EXTRA_FLAGS="$BITCODE_FLAGS -miphoneos-version-min=$SDK_IOS_MIN_VERSION"
             TARGET_SUFFIX=""
             ;;
         (iPhoneSimulator)
-            ARCH=i386
-            MARCHS="i386 x86_64 arm64"
-            EXTRA_FLAGS="$BITCODE_FLAGS -miphoneos-version-min=$SDK_IOS_MIN_VERSION"
+            ARCH=x86_64
+            MARCHS="x86_64 arm64"
+            EXTRA_FLAGS="$BITCODE_FLAGS -miphonesimulator-version-min=$SDK_IOS_MIN_VERSION"
             TARGET_SUFFIX="-simulator"
             ;;
     esac
@@ -199,7 +207,7 @@ for TARGET in $TARGETS; do
     for MARCH in $MARCHS; do
 
         echo "building for $TARGET - $MARCH"
-        build_target &
+        build_target "$TARGET" "$MARCH" "$ARCH" "$SYSROOT" "$EXTRA_FLAGS" "$TARGET_SUFFIX" &
         pid="$!"
         pids="$pids $pid"
         if test "x$parallel_mode" != x1 ; then

--- a/build-mac/libetpan.xcodeproj/project.pbxproj
+++ b/build-mac/libetpan.xcodeproj/project.pbxproj
@@ -2590,7 +2590,7 @@
 					"$(SRCROOT)/libsasl-ios/include",
 				);
 				INSTALL_PATH = /usr/local/lib;
-				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.6;
 				LIBRARY_SEARCH_PATHS = "$(SRCROOT)/libsasl-ios/lib";
 				PRODUCT_NAME = "etpan-ios";
 				SDKROOT = iphoneos;
@@ -2611,7 +2611,7 @@
 					"$(SRCROOT)/libsasl-ios/include",
 				);
 				INSTALL_PATH = /usr/local/lib;
-				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.6;
 				LIBRARY_SEARCH_PATHS = "$(SRCROOT)/libsasl-ios/lib";
 				PRODUCT_NAME = "etpan-ios";
 				SDKROOT = iphoneos;

